### PR TITLE
Issue #1676 - Move remove Repositories, RepositoryCredentials, HelmCredentials settings from ArgoCDSettings structure

### LINKED ISSUE #1676: Argo CD reformats own ConfigMap which make it is difficult to manage Argo CD declaratively
**Describe the bug**
Argo CD reformats own ConfigMap while updating settings. This makes it is difficult to manage it argocd deployment declaratively.

**To Reproduce**
If we cannot reproduce, we cannot fix! Steps to reproduce the behavior:
1. Deploy self-managed argocd and include repos into config map
2. Try to update settings imperatively (e.g. reset admin password)
3. See the config map reformatted

**Screenshots**
image

**Version** v1.0.1

### PR DESCRIPTION
**Pull request by alexmt**
Closes https://github.com/argoproj/argo-cd/issues/1676

### CODE REVIEW

File: util/db/repository_test.go [hunk 1]
```diff
@@ -24,9 +24,7 @@ func TestRepoURLToSecretName(t *testing.T) {
 }
 
 func Test_getRepositoryCredentialIndex(t *testing.T) {
-	argoCDSettings := settings.ArgoCDSettings{
-		RepositoryCredentials: []settings.RepoCredentials{{URL: "http://known"}},
-	}
+	repositoryCredentials := []settings.RepoCredentials{{URL: "http://known"}}
```
Comment (alexec) [inline util/db/repository_test.go]: does this test belong in this package anymore?

File: util/settings/settings.go [hunk 2]
```diff
@@ -254,6 +249,73 @@ func (mgr *SettingsManager) GetResourceOverrides() (map[string]v1alpha1.Resource
...
+	if repositoriesStr != "" {
+		err := yaml.Unmarshal([]byte(repositoriesStr), &repositories)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return repositories, nil
+}
+
+func (mgr *SettingsManager) SaveRepositories(repos []RepoCredentials) error {
```
Comment (alexec) [inline util/settings/settings.go]: not tested?

File: util/settings/settings.go [hunk 3]
```diff
@@ -282,53 +344,6 @@ func (mgr *SettingsManager) GetSettings() (*ArgoCDSettings, error) {
 	return &settings, nil
 }
 
-// MigrateLegacyRepoSettings migrates legacy (v0.10 and below) repo secrets into the v0.11 configmap
-func (mgr *SettingsManager) MigrateLegacyRepoSettings(settings *ArgoCDSettings) error {
```
Comment (alexec) [inline util/settings/settings.go]: did you mean to delete this func?

File: util/db/repository_test.go (hunk 1 above)
Comment (alexmt) [reply util/db/repository_test.go]: I think it does. It tests function defined in repository.go .

File: util/settings/settings.go (hunk 2 above)
Comment (alexmt) [reply util/settings/settings.go]: fixed

File: util/settings/settings.go (hunk 3 above)
Comment (alexmt) [reply util/settings/settings.go]: Yes. This was mostly required for argocd with legacy settings deployed at Intuit.

**Comment by alexec** [comment]
e2e failure looks real

**Comment by alexmt** [comment]
Thanks for review @alexec . Please take another look

**Review by alexec — APPROVED** [review]

**Pull request merged**